src/BilinearInterpolation.jl

```diff
@@ -0,0 +1,151 @@
+    end
+    return nothing
+end
+
+"""
+    interpolatebilinear!(
+        ftarget::AbstractArray{FT,N},
+        bilinear::B,
+        fsource::AbstractArray{FT,N},
+    ) where {FT,N,B}
+
+Interpolate `fsource`, defined on source grid, onto the target grid.
+The horizontal source and target grids are defined in `bilinear`.
+Here `fsource` is an N-dimensional array where the last two dimensions are assumed to be horizontal dimensions.
+
+For example, `fsource` can be of size `[n1, n2..., nx, ny]`, where `nx` and `ny` are the horizontal dimensions.
+Single horizontal level is also supported.
+The number of horizontal levels should be same for both source and target arrays.
+
+https://en.wikipedia.org/wiki/Bilinear_interpolation
+"""
+function interpolatebilinear!(
+    ftarget::AbstractArray{FT, N},
+    bilinear::B,
+    fsource::AbstractArray{FT, N},
+) where {FT, N, B}
+    @assert N ≥ 2
+    (leveldimssource..., nxs, nys) = size(fsource)
+    (leveldimstarget..., nxt, nyt) = size(ftarget)
+
+    (; sourcex, sourcey, targetx, targety, startx, starty) = bilinear
+    nsourcex, nsourcey, ntargetx, ntargety = get_dims(bilinear)
+
+    # ensure number of vertical levels are same between fsource and ftarget
+    @assert leveldimssource == leveldimstarget
+    # ensure horizontal dimensions of fsource match with horizontal source grid dimensions 
+    # in bilinear 
+    @assert nxs == nsourcex && nys == nsourcey
+    # ensure horizontal dimensions of ftarget match with horizontal target grid dimensions
+    # in bilinear
+    @assert nxt == ntargetx && nyt == ntargety
+
+    leveldims = leveldimssource
+    levelcidxs = CartesianIndices(leveldims)
+
+    @inbounds begin
+        for (iyt, y) in enumerate(targety)
+            sty = starty[iyt]
+            ly = sourcey[sty + 1] - sourcey[sty]
+            dy2 = sourcey[sty + 1] - y
+            dy1 = y - sourcey[sty]
+
+            for (ixt, x) in enumerate(targetx)
+                stx = startx[ixt]
+                lx = sourcex[stx + 1] - sourcex[stx]
+                dx2 = sourcex[stx + 1] - x
+                dx1 = x - sourcex[stx]
+                fac = FT(1) / (lx * ly)
+
+                for levelcidx in levelcidxs
+                    levelidx = Tuple(levelcidx)
+                    ftarget[levelidx..., ixt, iyt] =
+                        (
+                            dx2 * dy2 * fsource[levelidx..., stx, sty] +
+                            dx1 * dy2 * fsource[levelidx..., stx + 1, sty] +
+                            dx2 * dy1 * fsource[levelidx..., stx, sty + 1] +
+                            dx1 * dy1 * fsource[levelidx..., stx + 1, sty + 1]
+                        ) * fac
+                end
+            end
+        end
+    end
+    return nothing
+end
+
+"""
+    get_stencil_bilinear1d(alg::Linear, source, target; first = 1)
+
+Return the left index of location of `target` point in `source` grid.
+"""
+function get_stencil_bilinear1d(alg::Linear, source, target; first = 1)
```

Comment:
```suggestion
function get_stencil_leftidx(alg::Linear, source, target; first = 1)
```
(and docstring name too)

Reply:
Docstring name updated.